test/unittests/test_qfr_functionality.cpp

```diff
@@ -1937,3 +1937,106 @@ TEST_F(QFRFunctionality, dumpAndImportTeleportation) {
   ASSERT_EQ(qcImported.size(), 1);
   EXPECT_EQ(qcImported.at(0)->getType(), OpType::Teleportation);
 }
+
+TEST_F(QFRFunctionality, invertUnsupportedOperation) {
+  auto op = NonUnitaryOperation(1, Targets{0}, OpType::Measure);
+
+  ASSERT_THROW(op.invert(), QFRException);
+}
+
+TEST_F(QFRFunctionality, invertStandardOpSelfInverting) {
+  auto opTypes = {
+      OpType::I, OpType::X, OpType::Y, OpType::Z, OpType::H, OpType::SWAP,
+  };
+
+  for (auto opType : opTypes) {
```

Comment:
```suggestion
  for (const auto opType : opTypes) {
```
Same as above and throughout the remaining tests